feat: resizable modal

rahulramesha opened the pull request:
## Description

feature for resizable modal

Fixes #3850 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?


## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes







## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>


    // Code coverage diff between base branch:release and head branch: feature/resizableModal 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.75 **(-0.05)** | 36.51 **(-0.05)** | 33.89 **(-0.04)** | 55.3 **(-0.05)**
 :red_circle: | app/client/src/components/editorComponents/ResizeStyledComponents.tsx | 96.97 **(-3.03)** | 96 **(-4)** | 100 **(0)** | 100 **(0)**
 :red_circle: | app/client/src/resizable/index.tsx | 59.52 **(0.98)** | 36.96 **(-3.04)** | 35 **(0)** | 61.64 **(1.08)**
 :red_circle: | app/client/src/sagas/ModalSagas.ts | 35.71 **(-13.5)** | 17.65 **(-7.35)** | 23.08 **(-6.92)** | 36.67 **(-13.33)**
 :green_circle: | app/client/src/utils/DSLMigrations.ts | 75.04 **(0.17)** | 66.2 **(-0.07)** | 69.77 **(0)** | 74.64 **(0.18)**
 :sparkles: :new: | **app/client/src/utils/migrations/ModalWidget.ts** | **86.36** | **86.67** | **100** | **85.71**
 :green_circle: | app/client/src/widgets/ModalWidget/index.ts | 42.86 **(4.4)** | 0 **(0)** | 0 **(0)** | 42.86 **(4.4)**
 :red_circle: | app/client/src/widgets/ModalWidget/component/index.tsx | 25.86 **(-10.5)** | 9.76 **(-12.46)** | 0 **(0)** | 26.79 **(-9.57)**
 :red_circle: | app/client/src/widgets/ModalWidget/widget/index.tsx | 37.93 **(-5.55)** | 0 **(0)** | 17.65 **(-2.35)** | 37.35 **(-5.73)**</details>

rahulramesha commented:
/ok-to-test sha=bd9b882

marks0351 commented on app/client/src/sagas/WidgetOperationSagas.tsx:
```diff
@@ -477,6 +480,48 @@ const resizeCanvasToLowestWidget = (
+ * @param parentId
+ * @param height
+ */
+const getModalCanvasBottomRow = (
```
Can you check this resizeCanvasToLowestWidget coz getModalCanvasBottomRow looks like it can be extracted out of it. if not we can have two functions dedicated.

rahulramesha replied:
@marks0351 yes, i Initailly tried using that method but needed some changes specific to modal resize on `getModalCanvasBottomRow`. So I decided to have a dedicated method for that.

marks0351 commented on app/client/src/sagas/WidgetOperationSagas.tsx:
```diff
@@ -1864,6 +1958,7 @@ export default function* widgetOperationSagas() {
     ),
 
     takeLatest(WidgetReduxActionTypes.WIDGET_RESIZE, resizeSaga),
+    takeLatest(WidgetReduxActionTypes.WIDGET_MODAL_RESIZE, resizeModalSaga),
```
Also can we move this ModalSagas. it wud be more relevant.

marks0351 commented:
@rahulramesha also for some weird reason even in prod i cannot move a widget to the last column of modal canvas. can you check that too? as it moves to the last column it shows that the widget is out of bounds(red).

vivekverma2312 commented:
@rahulramesha Issue1 : Modal size changes but the widget size on Modal do not change on changing the view mode
https://www.loom.com/share/f33feb8ec7dd41a18f79dc52cac66e42

vivekverma2312 commented:
@rahulramesha 2. Confirm and cancel button is not of same size -- Is this expected (cc: @somangshu)
3. After disabling quick dismiss “on close “ property do not work

vivekverma2312 commented:
4. Not able to resize modal widget after placing List widget
https://www.loom.com/share/5459514765fa4c279e0c1a58b5e44aad

rahulramesha commented:
@marks0351 I have fixed the last column issue, please re-review
@vivekverma2312 fixed issues 1 and 4, 
issue 2, seems that it was part of configuration, i have modified it to have equal width from now on in new Modals
issue 3, seems to be an unrelated bug that is reproducible in production

vivekverma2312 commented:
@rahulramesha Above issues are verified and working fine. For issue 3 I have raised a separate git issue #7628
